provisioning/virtual-machines/ansible/roles/gns3/tasks/main.yml

```diff
@@ -75,3 +75,114 @@
     owner: "{{ ansible_user }}"
     group: "{{ ansible_user }}"
     mode: '0644'
+
+### The following Ansible tasks replace the previous manual setup of the GNS3 environment
+
+## 1.  Download the Cisco IOS Image
+
+# 1.1 Download nfs-common
+- name: Install nfs utils
+  apt:
+    name: nfs-common
+    state: present
+
+# 1.2 Mount GNS4 NFS install volume
+- name: Mount GNS3 NFS install volume
+  ansible.posix.mount:
+    src: "{{ gns3_device_images }}"
+    path: "/mnt/gns3_images"
+    opts: rw,sync,hard
+    boot: false
+    fstype: nfs
+    state: mounted
+
+# 1.3 Decompress the file from .bin to .image
+- name: Unzip the file
+  shell: >
+        unzip -p /mnt/gns3_images/c2691-advipservicesk9-mz.124-25c.bin >
+        /home/{{ ansible_user }}/GNS3/images/IOS/c2691-advipservicesk9-mz.124-25c.image
+  ignore_errors: yes
```

Comment:
It's always better to use native Ansible directives when possible, to avoid incompatibilities and possibly easing results.
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/unarchive_module.html

Reply:
Yes. I did modify the tasks according to what you suggested.
Thanks!